src/Workspaces/CSharp/Portable/Simplification/Reducers/CSharpMiscellaneousReducer.cs

```diff
@@ -110,14 +111,61 @@ public CSharpMiscellaneousReducer() : base(s_pool)
             OptionSet optionSet,
             CancellationToken cancellationToken)
         {
-            if (node.Statements.Count == 1 &&
-                CanHaveEmbeddedStatement(node.Parent) &&
-                !optionSet.GetOption(CSharpCodeStyleOptions.PreferBraces).Value)
+            if (node.Statements.Count != 1)
             {
-                return node.Statements[0];
+                return node;
             }
 
-            return node;
+            if (!CanHaveEmbeddedStatement(node.Parent))
+            {
+                return node;
+            }
+
+            switch (optionSet.GetOption(CSharpCodeStyleOptions.PreferBraces).Value)
+            {
+                case PreferBracesPreference.Always:
+                    return node;
+
+                case PreferBracesPreference.WhenMultiline:
+                    // Braces are optional in several scenarios for 'when_multiline', but are only automatically removed
+                    // in a subset of cases where all of the following are met:
+                    //
+                    // 1. This is an 'if' statement
+                    // 1. The 'if' statement does not have an 'else' clause and is not part of a larger 'if'/'else if'/'else' sequence
+                    // 2. The 'if' statement is not considered multiline
+                    if (!node.Parent.IsKind(SyntaxKind.IfStatement))
+                    {
+                        // Braces are only removed for 'if' statements
+                        return node;
+                    }
+
+                    if (node.Parent.IsParentKind(SyntaxKind.IfStatement, SyntaxKind.ElseClause))
+                    {
+                        // Braces are not removed from more complicated 'if' sequences
+                        return node;
+                    }
+
+                    if (!FormattingRangeHelper.AreTwoTokensOnSameLine(node.Statements[0].GetFirstToken(), node.Statements[0].GetLastToken()))
+                    {
+                        // Braces are not removed when the embedded statement is multiline
+                        return node;
+                    }
+
+                    if (!FormattingRangeHelper.AreTwoTokensOnSameLine(node.Parent.GetFirstToken(), node.GetFirstToken().GetPreviousToken()))
+                    {
+                        // Braces are not removed when the part of the 'if' statement preceding the embedded statement
+                        // is multiline.
+                        return node;
+                    }
+
+                    break;
+
+
+                case PreferBracesPreference.None:
+                    break;
+            }
+
+            return node.Statements[0];
```

Comment:
i would personally prefer this being inlined into the none/when-multiline cases.

Reply:
➡️ I intentionally wrote it so all the early return points mean the same thing.